### ISSUE TITLE
Registry: creating image stream pulls all tags in from docker.io

### ISSUE THREAD

**Issue by thomasmckay** [cockpit-project/cockpit#5116]
In UI I created an image stream _docker.io/thomasmckay/hammer_ and specified a specific tag _nightly_. However, both _latest_ and _nightly_ tags were pulled into the image stream.

OpenShift bug: https://bugzilla.redhat.com/show_bug.cgi?id=1373332

**Comment by stefwalter** [comment]
Could you document the exact actions that you used to run into this problem?

**Comment by thomasmckay** [comment]
Created new project, "dockerhub", with anonymous pull.
Created image stream named "hammer" with pull specific tags from "docker.io/thomasmckay/hammer" and specified tag "nightly".
Resulted in both "nightly" and "latest" tags appearing.

Note that I've been pulling that image and all the tags around into various places (eg. pulled into katello and then sync from there into atomic registry). Could it be that the tags are bleeding between projects? Just a guess.

**Comment by stefwalter** [comment]
This needs some diagnosis, to try and figure out where the Registry console differs in behavior with the ```oc``` command line tooling. Hopefully once we figure that out, it's an easy bug to fix.

**Comment by martinpitt** [comment]
I debugged that a while ago in https://bugzilla.redhat.com/show_bug.cgi?id=1373332 and added a reproducer with the `oc` command line tool. This isn't a Cockpit bug per se, but pulling specific tags only is underdocumented and quite possibly not even supported.

I also [wrote a test case](https://github.com/cockpit-project/cockpit/pull/6398/commits/07b603b1ffd691ff2a5ff41bc62328feb4839fd9) for this a while ago (commented out).

**Comment by thomasmckay** [comment]
Revisiting this issue. Below is the POST call from cockpit:
```

      /oapi/v1/namespaces/myproject/imagestreams?timeout=120s
      kind:"ImageStream"
      metadata:
        name:"alpinez"
        namespace:"myproject"
        spec:
          dockerImageRepository:"https://satellite.example.com:5000/examplecorp-development-alpineos-dockerhub-alpine"
          tags:[
            importPolicy:
              insecure:true
            name:"3.4"
```

This is a call that works w/ _oc_ tool
```
oc import-image alpine1:3.3 --from satellite.example.com:5000/examplecorp-dockerhub-alpine:3.3 --confirm --insecure=true --loglevel=10

      /oapi/v1/namespaces/myproject/imagestreamimports
      {
          "kind": "ImageStreamImport",
          "apiVersion": "image.openshift.io/v1",
          "metadata": {
              "name": "alpine1",
              "namespace": "myproject",
              "creationTimestamp": null
          },
          "spec": {
              "import": true,
              "images": [
                  {
                      "from": {
                          "kind": "DockerImage",
                          "name": "satellite.example.com:5000/examplecorp-dockerhub-alpine:3.3"
                      },
                      "to": {
                          "name": "3.3"
                      },
                      "importPolicy": {
                          "insecure": true
                      },
                      "referencePolicy": {
                          "type": ""
                      }
                  }
              ]
          },
          "status": {

          }
      }
```

**Comment by petervo** [comment]
Looked into this a little and according to the code comments in origin https://github.com/openshift/origin/blob/master/pkg/image/controller/imagestream_controller.go#L209. If we supply a dockerImageRepository on the spec all tags are always pulled. Looking at the history it's been like this for a long time.

Our UI is pretty much built around having one dockerImageRepository per imagestream but the actual origin implementation supports having a bunch of different ones mixed in together. So while I could fix to make this work fine on create to support edit we might need to make some changes.

My suggestion is that we drop the inline tag editing and the ""Pull specific tags from another image repository" options from the edit dialog. Instead we add an import image dialog, this would allow users to add any tag from any repo (with the appropriate refresh, and security settings) to the same image stream. We already have the ability to delete tags from image streams.

Otherwise we have to find another way to support tags from multiple repos without having a "default" we will also need to do more validation to resolve name conflicts.

@andreasn or @stefwalter any thoughts?

**Comment by martinpitt** [comment]
This got fixed in commit 2fae816b4e5c162333b5786c0e661315c0e49fdc
